libraries/SPI/SPI.h

```diff
@@ -25,6 +25,7 @@
 #include <stdlib.h>
 
 #define SPI_HAS_TRANSACTION
+#define SPI_SKIP_INITIAL_BUSY_CHECK //comment this line for backward compatibility if really needed
```

Comment:
I would say, backwards compatibility should be the default option. Making people modify library source code to achieve compatibility is not very nice (especially for a library bundled with the core, which is installed by the IDE who knows where...).

Actually, we only need to wait for SPI to become !busy if the current transfer uses different settings (frequency, data size, data order, etc) from the previous one. So I suggest to move the decision of waiting or not for the transfer to the run time.
So maybe add a member function like `_waitForSPINotBusy` which would compare new settings to the ones used during previous transfer and either return (if they are the same) or wait for SPI to be !busy (if they are different).